### PR TITLE
Add `g` to toggle between staging and workdir (`workarea`)

### PR DESCRIPTION
**Pull request by terhechte**
As discussed in https://github.com/extrawurst/gitui/issues/594 This adds `g` to toggle between staging and workdir. Needed to find a good term, went with `work area` which would be either staging or workdir. Open for suggestions :)

Note that the `vim` key config uses `w` instead of `g` as that (kinda) matches the vim `w` for moving between windows. Whereas `g` in vim can be used for jumping to the top. This isn't used yet but I'd love to add it here in the future.

### CODE REVIEW

File: src/tabs/status.rs [hunk 1]
```diff
@@ -236,6 +247,10 @@ impl Status {
         self.focus == Focus::Diff
     }
 
+    const fn can_toggle_workarea(&self) -> bool {
+        !matches!(self.focus, Focus::Diff)
```
Comment (extrawurst) [inline src/tabs/status.rs:251]: let's reuse `!is_focus_on_diff()`

File: src/strings.rs [hunk 2]
```diff
@@ -503,6 +503,18 @@ pub mod commands {
             CMD_GROUP_GENERAL,
         )
     }
+    pub fn toggle_workarea(
```
Comment (extrawurst) [inline src/strings.rs:506]: lets use two explicit methods here, one for `status_focus_workdir` and `status_focus_stage` then we can go back to the short terms `To Stage`/`To Unstated` like it was before (see https://github.com/extrawurst/gitui/pull/575).

File: src/tabs/status.rs [hunk 3]
```diff
@@ -515,6 +530,16 @@ impl Component for Status {
                 self.can_focus_diff(),
                 (self.visible && !focus_on_diff) || force_all,
             ));
+            out.push(
+                CommandInfo::new(
```
Comment (extrawurst) [inline src/tabs/status.rs:534]: lets use two different commands depending on where the focus is (see above)

**Review by extrawurst — CHANGES_REQUESTED** [review]
added a few comments.

**Comment by extrawurst** [comment]
how about using `w` in regular key bindings aswell, now that we have not strong reason to use `g` anyway?

**Comment by terhechte** [comment]
Good Point ☺️

**Comment by savchenko** [comment]
Would it make sense to allow <kbd>w</kbd> from the Diff view as well?
It currently does nothing and the only way to switch back is via <kbd>&larr;</kbd>.

**Comment by extrawurst** [comment]
I do not think this would we intuitive. I prefer the current situation where you always go back and forth with `left`/`right` between the file and the matching diff of that file

**Pull request merged**